faf-commons-data/src/main/java/com/faforever/commons/replay/header/ReplayHeaderToken.java

```diff
@@ -0,0 +1,10 @@
+package com.faforever.commons.replay.header;
+
+import java.util.List;
+
+public record ReplayHeaderToken(String gameVersion, String replayVersion, String pathToScenario, boolean cheatsEnabled, int seed,
+                                List<Source> sources,
+                                byte[] mods,
+                                byte[] gameOptions,
+                                List<byte[]> playerOptions) {
+}
```

Comment:
Why did you decide to have the header be represented by this more raw value and a more complete value in ReplayHeader? I don't know if there is the same value to having both of these that there is for the body tokens.

Reply:
I was a little bit at a loss here. The header is not as nicely structured as the body of a replay. I did find a few cases where the header has a leading 'size' integer and I interpret those as 'I'll process whatever is in there during parsing'

Reply:
I think it is probably better to just parse all the fields directly rather than having two objects since there isn't exactly the same issue with new tokens and the size notation.

Reply:
That would forfeit the parsing phase; should I then just process/parse everything in the `ReplayHeaderParser` class?

Reply:
Yeah I would think so. Since it seems like the header is really more like a single set of bytes.